test/integration/puppetlabs/general_puppet/general_puppet_int_test.clj

```diff
@@ -76,50 +82,68 @@
-            (let [response (http-client/get "https://localhost:8140/puppet/v3/static_file_content/?code_id=foobar&environment=test"
-                                            (assoc testutils/ssl-request-options
-                                              :as :stream))]
-              (is (= 400 (:status response)))
-              (is (= error-message (slurp (:body response)))))))))
+      (let [error-message "Error: A /static_file_content request requires an environment, a code-id, and a file-path"]
+        (testing "the /static_file_content endpoint returns an error if code_id is not provided"
+          (let [response (get-static-file-content "modules/foo/files/bar.txt?environment=test")]
+            (is (= 400 (:status response)))
+            (is (= error-message (slurp (:body response))))))
+        (testing "the /static_file_content endpoint returns an error if environment is not provided"
+          (let [response (get-static-file-content "modules/foo/files/bar.txt?code_id=foobar")]
+            (is (= 400 (:status response)))
+            (is (= error-message (slurp (:body response))))))
+        (testing "the /static_file_content endpoint returns an error if file-path is not provided"
+          (let [response (get-static-file-content "?code_id=foobar&environment=test")]
+            (is (= 400 (:status response)))
+            (is (= error-message (slurp (:body response)))))))
+      (let [error-message "Request Denied: A /static_file_content request must be a file within the files directory of a module or a module in an environment"
+            valid-paths ["modules/foo/files/bar.txt"
+                         "modules/foo/files/bar/"
+                         "modules/foo/files/bar/baz.txt"
+                         "environments/production/modules/foo/files/bar.txt"
+                         "environments/production/modules/foo/files/..conf..d../..bar..txt.."
+                         "environments/test/modules/foo/files/bar/baz.txt"
+                         "environments/dev/modules/foo/files/path/to/file/something.txt"]
+            invalid-paths ["modules/foo/manifests/bar.pp"
+                           "manifests/site.pp"
+                           "environments/foo/bar/files"
+                           "environments/../manifests/files/site.pp"
+                           "environments/production/files/~/.bash_profile"
+                           "environments/../modules/foo/files/site.pp"
+                           "environments/../modules/foo/lib/puppet/parser/functions/site.rb"
+                           "environments/production/modules/foo/files/../../../../../../site.pp"]]
+        (testing "valid requests return 200 and the file"
+          (doseq [path valid-paths]
+            (let [response (get-static-file-content (str path "?code_id=foobar&environment=production"))]
+              (is (= 200 (:status response)))
```

Comment:
We tend to avoid assertions inside a `doseq` like this, instead preferring to duplicate the necessary parts of the test.  It makes the test more verbose and repetitive, but it's much easier to figure out exactly what's going on when an assertion fails.